core/src/main/scala/org/apache/spark/deploy/worker/WorkerArguments.scala

```diff
@@ -23,46 +23,56 @@ import org.apache.spark.util.{IntParam, MemoryParam, Utils}
 import org.apache.spark.SparkConf
 
 /**
- * Command-line parser for the worker.
+ * Arguments parser for the worker.
  */
 private[spark] class WorkerArguments(args: Array[String], conf: SparkConf) {
-  var host = Utils.localHostName()
-  var port = 0
-  var webUiPort = 8081
-  var cores = inferDefaultCores()
-  var memory = inferDefaultMemory()
+  var host: String = null
+  var port: Int = -1
+  var webUiPort: Int = -1
+  var cores: Int = -1
+  var memory: Int = -1
```

Comment:
How could we know whether `port` is already assigned with the value after `--port`(in the function `parse`)?